### PR TITLE
zh-cn: update the translation of "Relationship of flexbox to other layouts"

### PR DESCRIPTION
**Pull request by yin1999**
### Description

update the translation of "Relationship of flexbox to other layout methods"

### Related issues and pull requests

Blocked by: mdn/content#32605
Part of: #14272

### CODE REVIEW

**Comment by yin1999** [comment]
We have been waiting for the upstream fixes for a quite long time. Let's get this merged first, as the original content is too outdated.

File: files/zh-cn/web/css/css_flexible_box_layout/relationship_of_flexbox_to_other_layout_methods/index.md [hunk 1]
```diff
@@ -1,39 +1,37 @@
...
+这些属性最开始出现在弹性盒子规范中，现在已经成为[盒子对齐规范](https://www.w3.org/TR/css-align-3/)的一部分。这个规范详细说明了在所有布局中（不仅仅是弹性盒子）对齐属性是如何起作用的。盒子对齐用于处理对齐方式以及内容调整（包括在弹性元素之间创建间隙或上下间距）。
 
-之所以在 flexbox 规范和 box alignment 模块规范中都有对对齐属性的详细描述，是为了确保 flexbox 规范的完成不会受 box alignment 模块规范的影响，因为后者需要详细说明所有的布局类型中的对齐方法。flexbox 规范中有一条注释指出将来一旦 Box Alignment Level 3 完成，它将会取代 flexbox 规范中的相关定义：
+之所以在弹性盒子规范和盒子对齐规范中都有有关盒子对齐属性的详细描述，是为了确保弹性盒子规范的完成不会受盒子对齐规范的影响，因为后者需要详细说明所有的布局类型中的对齐方法。弹性盒子规范中有一条注释指出：将来，弹性盒子第 3 版规范一旦完成，它将会取代弹性盒子规范中的相关定义：
 
-> “注意：虽然对齐属性是在 CSS Box Alignment \[CSS-ALIGN-3] 中定义的，但“Flexible Box Layout”在此处重现了相关属性的定义，以免形成规范性的依赖关系，而这可能会减慢规范的发展。这些属性仅适用于 Flex 布局，直到 CSS Box Alignment Level 3 完成并定义其对其他布局模式的效果；此外，在 Box Alignment 模块中定义的任何新值都将应用于 Flexible Box Layout；换句话说，一旦 Box Alignment 模块完成，其中的相关定义将取代此处的定义。”
+> “注意：虽然对齐属性是在 CSS 弹性盒子 \[CSS-ALIGN-3] 中定义的，但弹性盒子布局在此处重现了相关属性的定义，以免形成规范性的依赖关系，而这可能会减慢规范的发展。在 CSS 弹性盒子第 3 级规范完成并定义其对其他布局模式的效果之前，这些属性仅适用于弹性布局。此外，在盒子对齐模块中定义的任何新值都将适用于弹性盒子布局；换句话说，一旦盒子对齐模块完成，其中的相关定义将取代此处的定义。”
 
-在本系列的后续文章（在 flex 容器中对齐元素）中，我们将彻底研究 Box Alignment 属性如何应用于 flex 元素。
+在本系列的后续文章——[在弹性容器中对齐元素](/zh-CN/docs/Web/CSS/CSS_flexible_box_layout/Aligning_items_in_a_flex_container)——中，我们将彻底研究 盒子对齐属性如何应用于弹性元素。
```
Comment (jasonren0403) [inline files/zh-cn/web/css/css_flexible_box_layout/relationship_of_flexbox_to_other_layout_methods/index.md:24]:
```suggestion
在本系列的后续文章——[在弹性容器中对齐元素](/zh-CN/docs/Web/CSS/CSS_flexible_box_layout/Aligning_items_in_a_flex_container)——中，我们将全面了解盒子对齐属性如何应用于弹性元素。
```

File: files/zh-cn/web/css/css_flexible_box_layout/relationship_of_flexbox_to_other_layout_methods/index.md [hunk 2]
```diff
@@ -43,66 +41,66 @@ The writing modes specification defines the following values of the {{cssxref("w
...
+弹性盒子规范包含了对如果一个元素使用了其他布局方式然后变成了弹性元素的这种情况[会发生什么的定义](https://www.w3.org/TR/css-flexbox-1/#flex-containers)。例如，如果一个元素是浮动的，那么它的父元素就会变成弹性容器。或者，弹性容器在不同布局中的行为。
 
-An element set to `display: flex` behaves in most ways like any other block level container that establishes a containing block. Floats will not intrude, and the containers' margins will not collapse.
+设置了 `display: flex` 的元素在大多数情况下都会像其他块级容器一样，它会创建一个包含块。浮动不会侵入，容器的外边距也不会折叠。
 
-With regard to flex items, if an item was floated or cleared and then becomes a flex item due to the parent having `display: flex` applied, the floating and clearing will no longer happen, and the item will not be taken out of normal flow in the way that floats are. If you have used the {{cssxref("vertical-align")}} property, as used with `inline-block` or table layout for alignment, this will no longer affect the item and you can use the alignment properties of flexbox instead.
+对于弹性元素，如果元素是浮动的或者清除了浮动，那将由于父元素应用了 `display: flex` 而变成弹性元素，浮动和清除也将不再发生，元素也不会像浮动那样脱离常规流。如果你使用了 {{cssxref("vertical-align")}} 属性，就像与 `inline-block` 或表格布局一起使用来进行对齐，这将不再影响该元素而且你可以改用弹性盒子的对齐属性。
```
Comment (jasonren0403) [inline files/zh-cn/web/css/css_flexible_box_layout/relationship_of_flexbox_to_other_layout_methods/index.md:54]:
```suggestion
对于弹性元素，如果元素是浮动的或者清除了浮动，然后因父元素应用了 `display: flex` 而变成弹性元素，那么浮动和清除也将不再发生，该元素也不会像浮动元素那样脱离常规流。如果你使用了 {{cssxref("vertical-align")}} 属性（与 `inline-block` 或表格布局一起用于对齐），这将不再影响该元素，可以改用弹性盒子的对齐属性来代替。
```

File: files/zh-cn/web/css/css_flexible_box_layout/relationship_of_flexbox_to_other_layout_methods/index.md [hunk 3]
```diff
@@ -43,66 +41,66 @@ The writing modes specification defines the following values of the {{cssxref("w
...
+通过为嵌套元素的包装器添加 `display: contents`，你可以看到该元素已从布局中消失，从而允许两个子元素被布局，就像它们是弹性容器的直接子元素一样。你可以尝试移除 `display: contents` 这行来查看其结果。
 
-Note that this only removes the box from the layout; the sub-children don't become direct children in any other way. You can see that as I have used a direct child selector to add the background and borders to the flex items, this has not been applied to our nested children. They have been laid out as flex items, but as they are not direct children they do not get the other styling.
+请注意，这只会从布局中删除该盒子；子元素并不会以任何其他方式成为直接子元素。你可以看到，我们使用了直接子元素选择器来为弹性元素添加背景和边框，因此这些样式并没有应用到嵌套的子元素上。它们已经被布局为弹性元素，但由于它们不是直接子元素，所以它们不会获得其他样式。
 
-> **警告：** Use of `display: contents` will also remove the element from the accessibility tree – screen readers will not see what's inside, just the same as if you used `display: none`. Use of `contents` should only be for presentational, not content, elements.
+> **警告：** 大多数浏览器的当前实现会从可访问树中移除任何具有 `display: contents` 的元素（但会保留其后代）。这将导致屏幕阅读技术不再宣告该元素。根据规范，这是不正确的行为，参见 [`display: contents`](/zh-CN/docs/Web/CSS/display#display_contents)。
```
Comment (jasonren0403) [inline files/zh-cn/web/css/css_flexible_box_layout/relationship_of_flexbox_to_other_layout_methods/index.md:100]:
```suggestion
> **警告：** 大多数浏览器的当前实现会从无障碍树中移除任何具有 `display: contents` 的元素（但会保留其后代）。这将导致屏幕阅读技术不再宣告该元素。根据规范，这是不正确的行为，参见 [`display: contents`](/zh-CN/docs/Web/CSS/display#display_contents)。
```

**Review by jasonren0403 — COMMENTED** [review]
nits

**Review by jasonren0403 — APPROVED** [review]
👍

**Pull request merged**